packages/slice-machine/lib/models/common/widgets/ContentRelationship/index.ts

```diff
@@ -60,4 +62,21 @@ export const ContentRelationshipWidget: Widget<Link, typeof schema> = {
   FormFields,
   CUSTOM_NAME: "ContentRelationship",
   Form,
+  prepareInitialValues: (initialValues) => {
+    const customTypes = useSelector(selectAllCustomTypes);
```

Comment:
i don't know if using the selector is a good idea here because it's not in a component or anything, but i understand that it's morel logical to have it here.

Reply:
I think if we want to keep generic things generic, then it seems like the other options are:

1. Do this filtering in the widget form like we had before
2. Have a selector which gets the whole state and passes it to the prepareInitialValues of all the forms (seems excessive)
3. Have the selector in here

It doesn't seem to be causing any issues I think. What do you recon?

Reply:
It seems good then.
I was just worried because I usually see those at the top level of a react component.
Maybe i'm too narrow minded on this one 😄